etl/kpinetwork_analizer.py

```diff
@@ -82,6 +82,138 @@
+        },
+    )
+    dynamic_frame = dynamic_frame.toDF().createOrReplaceTempView(
+        "{}_table".format(db_table)
+    )
+    sqldf = spark.sql("SELECT id FROM {}_table".format(db_table))
+    result_from_sqldf = sqldf.collect()
+    ids_from_db = [row.id for row in result_from_sqldf]
+    return ids_from_db
+
+
+def read_table_from_database(env, database, db_table):
+    ctg_connection = "{}_connection".format(env)
+    catalog_connection = glueContext.extract_jdbc_conf(ctg_connection)
+
+    return glueContext.create_dynamic_frame.from_options(
+        connection_type="postgresql",
+        connection_options={
+            "url": "{}/{}".format(catalog_connection["url"], database),
+            "user": catalog_connection["user"],
+            "password": catalog_connection["password"],
+            "dbtable": db_table,
+        },
+    )
+
+
+def create_temp_table(dynamic_frame, table_name):
+    return dynamic_frame.toDF().createOrReplaceTempView(table_name)
+
+
+def existing_metric_from_file(env, company_id, scenario_name, metric_name):
+    resultant_metric_id = None
+    database = "kpinetworkdb"
+    if env == "demo":
+        database = "demokpinetworkdb"
+
+    dynamic_scenario_frame = read_table_from_database(
+        env, database, "financial_scenario"
+    )
+    dynamic_scenario_frame = create_temp_table(
+        dynamic_scenario_frame, "financial_scenario_table"
+    )
+    scenario_dataframe = spark.sql(
+        "SELECT id FROM financial_scenario_table WHERE company_id = '{}' AND name = '{}'".format(
+            company_id, scenario_name
+        )
+    )
+    result_scenario_dataframe = scenario_dataframe.collect()
+    scenario_ids = [row.id for row in result_scenario_dataframe]
+
+    if not scenario_ids:
+        return resultant_metric_id
+
+    financial_scenario_id = scenario_ids[0]
+    dynamic_sm_frame = read_table_from_database(env, database, "scenario_metric")
+    dynamic_sm_frame = create_temp_table(dynamic_sm_frame, "scenario_metric_table")
+    sm_dataframe = spark.sql(
+        "SELECT metric_id FROM scenario_metric_table WHERE scenario_id = '{}'".format(
+            financial_scenario_id
+        )
+    )
+    result_sm_dataframe = sm_dataframe.collect()
+    sm_ids = [row.metric_id for row in result_sm_dataframe]
+
+    if not sm_ids:
+        return resultant_metric_id
+
+    sc_metric_ids = sm_ids
+    for sc_metric_id in sc_metric_ids:
+        metric_id = sc_metric_id
+        dynamic_m_frame = read_table_from_database(env, database, "metric")
+        dynamic_m_frame = create_temp_table(dynamic_m_frame, "metric_table")
+        m_dataframe = spark.sql(
+            "SELECT * FROM metric_table WHERE id = '{}' AND name = '{}'".format(
+                metric_id, metric_name
+            )
+        )
+        result_m_dataframe = m_dataframe.collect()
+        m_ids = [row for row in result_m_dataframe]
+
+        if m_ids != []:
+            resultant_metric_id = m_ids
+            break
```

Comment:
Changf if condition "m_ids != []" to "m_ids"

Reply:
changed condition.